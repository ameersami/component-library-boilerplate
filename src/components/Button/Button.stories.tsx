import React from 'react';
import { addParameters } from '@storybook/react';
import { withA11y } from '@storybook/addon-a11y';
import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';
import Button from './Button';

addParameters({
  viewport: {
    viewports: INITIAL_VIEWPORTS,
  },
});

addParameters({
  backgrounds: [
    { name: 'twitter', value: '#00aced' },
    { name: 'facebook', value: '#3b5998' },
  ]
})

export default {
  title: 'button',
  decorators: [withA11y],
};

export const inaccessible = () => (
  <Button message="Inaccessible button"/>
);