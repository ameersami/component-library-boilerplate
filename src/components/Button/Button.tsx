import React from 'react';
import propTypes, { InferProps } from 'prop-types';

const ButtonPropTypes = {
  message: propTypes.string
}

const ButtonComponent = (props: InferProps<typeof ButtonPropTypes>) => (
  <div>
    <span>{props.message}</span>
  </div>
);
ButtonComponent.propTypes = ButtonPropTypes;
ButtonComponent.defaultProps = { message: '' }

export default ButtonComponent;