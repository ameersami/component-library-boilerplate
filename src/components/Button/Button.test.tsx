import * as React from 'react';
import {shallow} from 'enzyme';

import Button from './Button';

describe('<Button/>', () => {
  test('Should render', () => {
    const wrapper = shallow(<Button/>);
    expect(wrapper).toMatchSnapshot();
  })
})