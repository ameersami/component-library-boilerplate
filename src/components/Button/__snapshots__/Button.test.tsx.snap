// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Should render 1`] = `
<div>
  <span />
</div>
`;
